plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -10,58 +10,65 @@ import android.content.ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE
 import android.content.ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN
 import android.content.res.Configuration
 import android.view.View
-import androidx.lifecycle.Lifecycle
-import androidx.lifecycle.LifecycleObserver
-import androidx.lifecycle.OnLifecycleEvent
-import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.lifecycle.*
 import com.mapbox.maps.MapboxLifecycleObserver
-import com.mapbox.maps.logE
 import com.mapbox.maps.logW
 import com.mapbox.maps.plugin.Plugin.Companion.MAPBOX_LIFECYCLE_PLUGIN_ID
 import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
 
 /**
  * Concrete implementation of MapboxLifecyclePlugin.
  */
-class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
+class MapboxLifecyclePluginImpl constructor(mapView: View?) : MapboxLifecyclePlugin,
+  LifecycleOwner {
```

Comment:
A bit confusing that MapboxLifecyclePluginImpl is LifecycleOwner, returning instance of ViewLifecycleRegistry for the ViewLifecycleRegistry. Could ViewLifecycleRegistry implement LifecycleOwner itself instead? Code should become simpler then

Reply:
E.g. 
```
internal class ViewLifecycleOwner(
  view: View,
  hostingLifecycleOwner: LifecycleOwner,
) : LifecycleOwner {

  private val lifecycleRegistry = LifecycleRegistry(this)

  override fun getLifecycle() = lifecycleRegistry
```

Reply:
`ViewLifecycleRegistry` has to implement the `LifecycleRegistry`, in order to manipulate the lifecycle state, and merge the lifecycle with the view's detach/attach state.

Reply:
However pushed a commit to use a dummy `locaLifecycleOwner` instead of making `MapboxLifecyclePluginImpl` a `LifecycleOwner`

Reply:
> `ViewLifecycleRegistry` has to implement the `LifecycleRegistry`, in order to manipulate the lifecycle state, and merge the lifecycle with the view's detach/attach state.

Shouldn't be able to update the state `private val lifecycleRegistry = LifecycleRegistry(this)` should be used, so instead of `state = ` it becomes `lifecycleRegistry.state = `

Reply:
So it becomes : 
```
internal class ViewLifecycleOwner(
  view: View,
  hostingLifecycleOwner: LifecycleOwner,
) : LifecycleOwner {

  private val viewLifecycleRegistry = LifecycleRegistry(this)
  override fun getLifecycle() = viewLifecycleRegistry

  private var isAttached = view.isAttachedToWindow

  private val hostingLifecycleObserver = LifecycleEventObserver { _, event ->
    val isAtLeastCreated = currentState.isAtLeast(State.CREATED)
    if (isAttached || (isAtLeastCreated && event == Event.ON_DESTROY)) {
      viewLifecycleRegistry.handleLifecycleEvent(event)
    }
  }

  private val attachStateChangeListener = object : View.OnAttachStateChangeListener {
    override fun onViewAttachedToWindow(p0: View?) {
      viewLifecycleRegistry.currentState = hostingLifecycleOwner.lifecycle.currentState
      isAttached = true
    }

    override fun onViewDetachedFromWindow(p0: View?) {
      isAttached = false
      if (hostingLifecycleOwner.lifecycle.currentState.isAtLeast(State.STARTED)) {
        lifecycleRegistry.currentState = State.CREATED
      }
    }
  }

  init {
    hostingLifecycleOwner.lifecycle.addObserver(hostingLifecycleObserver)
    view.addOnAttachStateChangeListener(attachStateChangeListener)
  }
}
```

Reply:
pushed a commit with suggested changes.